html_css_dom/the_internet/README.md

```diff
@@ -0,0 +1,183 @@
+
+A `server` is a computer, listening for requests, and sending back responses. This response information can be many things, but is usually something like:
+
+- files: photos, js, css, html, or videos
+- data from a database
+
+A `client` is a computer that asks the server for information. A client can be one of many things - a laptop, a phone, a web browser, a program on your computer, a `node` script.
+
+You can run a `node` server on your computer and make requests to it from your same computer. We will do this a lot when we get into module 4. Where the server is and where the request comes from doesn't make any difference.
+
+## Request-Response lifecycle
+
+When we talk about requests in this context, we're specifically talking about [HTTP Messages](https://developer.mozilla.org/en-US/docs/Web/HTTP/Messages). A message is either a request or a response.
+
+![request response](./assets/request-response.jpg)
+
+- The client sends a request for data to the server.
+- The server processes that request and then sends a response back.
+- The client handles that response and then does something with the data, like displaying it on the screen - usually updating some HTML to make it appear.
+
+Note that the client is always the one that initiates the request. By definition, that's what makes it a client. The server is the one that responds to that request.
+
+**HTTP Status Codes**
+
+HTTP messages have special [Status Codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) that indicate whether the response was successful or not. 
+
+Status codes range from 100-500ish and are broken down into categories:
+
+- 100 - Information
+- 200 - Success
+- 300 - Redirection
+- 400 - Client errors
+- 500 - Server errors
+
+The most common status codes you'll see are:
+
+- `200` which means "OK", or that the request was successful and the response was sent back without issue.
+- `301` which means "this was moved elsewhere". If you type `google.com` and your browser takes you to `www.google.com` its because of a `301` response.
+- `404` which means "this page was not found"
+- `500` which means "the server encountered an error". Sometimes you get a nice message back about why, but often you do not.
+
+**HTTP Request Methods**
+
+There are multiple types of requests that clients can make. 
+
+These are known sometimes as HTTP "verbs". You can think of each one as an action.
+
+* GET - give me some data
+* POST - take this data
+* PUT - update this data
+* DELETE - delete this data
+
+
+## Making your own requests
+
+You may not realize that you have already been making requests!
+
+Every time you load a web page, some requests are made. We can look at all of the requests using the browser's devtools network tab.
+
+* Open any website, like the [main curriculum page](https://github.com/joinpursuit/Pursuit-Core-Web)
+* Open the devtools (Cmd-Shift-i)
```

Comment:
I think it's Cmd-Option-i

Reply:
will update! vivaldi and chrome are different, it turns out.

Reply:
TIL about vivaldi browser